gpfit/xfoil_wrapper.py

```diff
@@ -0,0 +1,124 @@
+" python wrapper to call xfoil "
+import subprocess
+import numpy as np
+
+#pylint: disable=invalid-name, bare-except, too-many-locals
+#pylint: disable=too-many-arguments
+
+def xfoil_comparison(airfoil, Cl, Re, Cd):
+    """
+    comparison of XFOIL cd to input cd for given cl and re
+    INPUTS
+    ------
+    airfoil:                    airfoil name
+                                    str - ("xxx.dat", "naca xxxx")
+    Cl:                         lift coefficient
+                                    float or 1d-list or array
+    Re:                         Reynolds number
+                                    float or 1d-list or array
+    Cd:                         drag coefficient
+                                    float or 1d-list or array
+
+    OUTPUTS
+    -------
+    err:                        error between Cd and XFOIL computed Cd
+                                    1-d numpy array
+    cdxs:                       XFOIL computed Cd values
+                                    1-d numpy array
+    """
+
+    if ".dat" in airfoil:
+        topline = "load " + airfoil + " \n afl \n"
+    elif "naca" in airfoil:
+        topline = airfoil + "\n"
+    else:
+        raise ValueError("Invalid airfoil. Valid types are *.dat, naca xxxx")
+
+    if not hasattr(Cl, "__len__"):
+        Cl, Re, Cd = [Cl], [Re], [Cd]
+
+    err, cdxs = [], []
+    for cl, re, cd in zip(Cl, Re, Cd):
+        failmsg = "Xfoil call failed at CL=%.4f and Re=%.1f" % (cl, re)
+        try:
+            x = single_call(topline, cl, re, 0.0)
+            if "VISCAL:  Convergence failed" in x:
+                print "Convergence Warning: %s" % failmsg
+                cdx = cd
+            else:
+                cdx = x[0]
+        except:
```

Comment:
danger Will Robinson. What are the kinds of exceptions that can be caused here? Probably SystemError and WindowsError?